inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/project/EvaluationSimulationPanel.java

```diff
@@ -0,0 +1,215 @@
+import de.tudarmstadt.ukp.clarin.webanno.model.SourceDocument;
+import de.tudarmstadt.ukp.clarin.webanno.security.UserDao;
+import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaAjaxButton;
+import de.tudarmstadt.ukp.inception.recommendation.api.RecommenderFactoryRegistry;
+import de.tudarmstadt.ukp.inception.recommendation.api.evaluation.EvaluationResult;
+import de.tudarmstadt.ukp.inception.recommendation.api.evaluation.IncrementalSplitter;
+import de.tudarmstadt.ukp.inception.recommendation.api.model.Recommender;
+import de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationEngine;
+import de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationEngineFactory;
+import de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationException;
+import de.tudarmstadt.ukp.inception.recommendation.chart.ChartPanel;
+import de.tudarmstadt.ukp.inception.recommendation.model.LearningCurve;
+
+public class EvaluationSimulationPanel
+    extends Panel
+{
+    private static final long serialVersionUID = 4306746527837380863L;
+
+    private static final String MID_CHART_CONTAINER = "chart-container";
+    private static final String MID_SIMULATION_START_BUTTON = "simulation-start-button";
+    private static final String MID_FORM = "form";
+    
+    private static final double TRAIN_PERCENTAGE = 0.8;
+    private static final int INCREMENT = 250;
+    private static final int LOW_SAMPLE_THRESHOLD = 10;
+
+    private static final Logger LOG = LoggerFactory.getLogger(EvaluationSimulationPanel.class);
+    
+    private @SpringBean DocumentService documentService;
+    private @SpringBean UserDao userDao;
+    private @SpringBean RecommenderFactoryRegistry recommenderRegistry;
+
+    private final ChartPanel chartPanel;
+    private final Project project;
+    private final IModel<Recommender> selectedRecommenderPanel;
+
+    public EvaluationSimulationPanel(String aId, Project aProject,
+            IModel<Recommender> aSelectedRecommenderPanel)
+    {
+        super(aId);
+        project = aProject;
+        selectedRecommenderPanel = aSelectedRecommenderPanel;
+                
+        Form<Recommender> form = new Form<>(MID_FORM);
+        add(form);
+        
+        //initially the chart is empty. passing empty model
+        chartPanel = new ChartPanel(MID_CHART_CONTAINER, Model.of());
+        chartPanel.setOutputMarkupId(true);
+        form.add(chartPanel);
+
+        // clicking the start button the annotated documents are evaluated and the learning curve
+        // for the selected recommender is plotted in the hCart Panel
+        @SuppressWarnings({ "unchecked", "rawtypes" })
+        LambdaAjaxButton startButton = new LambdaAjaxButton(MID_SIMULATION_START_BUTTON,
+            (_target, _form) -> {
+                String[] scoresAndTrainingSizes = evaluate(_target);
+                
+                if (scoresAndTrainingSizes == null) {
```

Comment:
If the scores are empty, there is a warning - why is there no message/warning if the scores are null? Can you please add explanatory comments here?

Reply:
Added comment. The warning has been shown in the method evaluate(_target) already. There are different scenarios when the score is returned null and each one is handled differently in the evaluate(_target).